tests/test_odcs_requester.py

```diff
@@ -0,0 +1,83 @@
+"""test_odcs_requester.py - test odcs_requester"""
+from typing import Callable
+
+# pylint: disable=redefined-outer-name
+from unittest.mock import MagicMock, create_autospec
+
+import pytest
+from odcs.client.odcs import ODCS, ComposeSourceTag  # type: ignore
+
+from generate_compose.odcs_configurations_generator import (
+    ODCSComposeConfig,
+    ODCSComposesConfigs,
+)
+from generate_compose.odcs_requester import ODCSRequester, ODCSRequestReferences
+
+
+@pytest.fixture()
+def compose_url() -> str:
+    """Example compose-url, as close to the one expected"""
+    return "http://download.eng.bos.redhat.com/odcs/prod/odcs-222222"
+
+
+@pytest.fixture()
+def create_odcs_mock(compose_url: str) -> Callable[[int], MagicMock]:
+    """Create an ODCS mock with specific results for the compose method"""
+
+    def _mock_odcs_compose(num_of_composes: int) -> MagicMock:
+        """Monkey-patched ODCS compose"""
+        mock: MagicMock = create_autospec(ODCS)
+        mock.request_compose.side_effect = [
+            {"result_repofile": f"{compose_url}", "id": i}
+            for i in range(1, num_of_composes + 1)
+        ]
+        return mock
+
+    return _mock_odcs_compose
+
+
+@pytest.fixture()
+def compose_source_tag() -> ComposeSourceTag:
```

Comment:
I think we should have a mock here of the generic type rather than an instance of one of its implementations.

Reply:
Maybe a fixture factory? either way we'll need to supply it with the `source_type`

Reply:
If we create a mock using `create_autospec` based on the the generic source, then I don't think we need anything.

Reply:
changed